### PR TITLE
feat/Wow Best Deals

### PR DESCRIPTION
**Pull request by krz0001**
## Why

Implements WoW Best deals page from the temp frontend.

## Related Tickets

* Closes https://github.com/ff14-advanced-market-search/saddlebag-with-pockets/issues/277

### CODE REVIEW

File: app/routes/wow._index.tsx [hunk 1]
```diff
@@ -95,6 +95,13 @@ const recommendedQueries = [
       'View the time each hour when the Blizzard API AH data updates.',
     Icon: DocumentSearchIcon,
     href: '/wow/upload-timers'
+  },
+  {
+    name: 'Best Deals',
+    description:
+      'Find the best deals on your server and region wide with our Best Deals search!',
+    Icon: DocumentSearchIcon,
+    href: '/wow/export-search'
```
[the remark this answers is not in the document]
Comment (maxencelav) [reply app/routes/wow._index.tsx:104]: Copy/pasted a bit too quick there 😅

[the remark this answers is not in the document]
Comment (krz0001) [reply app/routes/wow._index.tsx:104]: Copy/pasted a bit too quick there 😅 Doing a bit more fixes as it doesn't seem to appear in the sidebar for me.

File: app/requests/WoW/BestDeals.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+const WoWBestDeals = async ({
+  type,
+  region,
+  discount,
+  salesPerDay,
+  minPrice
+}: WoWBestDealsProps) => {
+  const floatingSalesPerDay = salesPerDay.toFixed(1)
+  console.log('floatingSalesPerDay', floatingSalesPerDay)
```
Comment (oldben87) [inline app/requests/WoW/BestDeals.ts:36]: Can remove the console log

File: app/routes/wow.best-deals.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  const region = session.getWoWSessionData().region
+
+  const formData = Object.fromEntries(await request.formData())
+
+  const validatedFormData = validateInput.safeParse(formData)
+  if (!validatedFormData.success) {
+    return json({ exception: 'Invalid Input' })
+  }
+
+  console.log({
```
Comment (oldben87) [inline app/routes/wow.best-deals.tsx:39]: Can remove the console.log

**Review by oldben87 — CHANGES_REQUESTED** [review]

**Comment by oldben87** [comment]
> LGTM, @oldben87 give it a quick look.  Im down to merge this in tomorrow once you have a chance to see it.

Once the console.logs are removed it looks good to merge!

**Review by oldben87 — APPROVED** [review]

**Pull request merged**